pkg/apis/workflow/v1alpha1/artifact_gc.go

```diff
@@ -0,0 +1,52 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:resource:shortName=agw
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+type ArtifactGCTask struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Spec              ArtifactGCSpec    `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status            ArtifactGCSStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+type ArtifactGCSpec struct {
+	ArtifactsByNode map[string]ArtifactNodeSpec `json:"artifactsByNode,omitempty" protobuf:"bytes,1,rep,name=artifactsByNode"`
+}
+
+type ArtifactNodeSpec struct {
+	ArchiveLocation *ArtifactLocation       `json:"archiveLocation,omitempty" protobuf:"bytes,1,opt,name=archiveLocation"`
+	Artifacts       map[string]ArtifactList `json:"artifacts,omitempty" protobuf:"bytes,2,rep,name=artifacts"`
+}
+
+type ArtifactList struct {
+	Artifacts `json:"artifacts" protobuf:"bytes,1,rep,name=artifacts"`
+}
+
+type ArtifactGCSStatus struct {
+	ArtifactResultsByNode map[string]ArtifactResultNodeStatus `json:"artifactResultsByNode,omitempty" protobuf:"bytes,1,rep,name=artifactResultsByNode"`
+}
+
+type ArtifactResultNodeStatus struct {
+	ArtifactResults []ArtifactResult `json:"artifactResults,omitempty" protobuf:"bytes,1,rep,name=artifactResults"`
+}
+
+type ArtifactResult struct {
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	Deleted bool `json:"deleted,omitempty" protobuf:"varint,2,opt,name=deleted"`
```

Comment:
will status make sense instead `deleted` ?
status=failed/deleted

Reply:
I guess it would be nice if we could still have a boolean type rather than a new type. Maybe "Failed" as a boolean? What do you think? We also have the "Error" there which will be set if it fails.

Reply:
I changed it to a "Success" boolean